gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/InMemoryGlobberFileSystem.java

```diff
@@ -105,7 +105,7 @@ public Path getWorkingDirectory() {
       throw new FileNotFoundException(
           String.format("Path '%s' (qualified: '%s') does not exist.", f, qualifiedPath));
     }
-    return fileStatuses.toArray(new FileStatus[0]);
+    return cloneFileStatusList(fileStatuses);
```

Comment:
This you can simply do something like:

`fileStatuses.stream().map(fileStatus -> new FileStatus(fileStatus)).collect(Collectors.toList());`

to save a method work.

Reply:
Made the clone method inline as per our discussion